internal/langserver/handlers/complete_test.go

```diff
@@ -844,14 +848,25 @@ output "test" {
 		}`)
 }
 
-func tfExecutor(t *testing.T, workdir, version string) exec.TerraformExecutor {
+func tfExecutor(t *testing.T, workdir, tfVersion string) exec.TerraformExecutor {
 	ctx := context.Background()
-	installPath := filepath.Join(t.TempDir(), "tfinstall")
-	err := os.MkdirAll(installPath, 0o755)
-	if err != nil {
-		t.Fatal(err)
-	}
-	execPath, err := tfinstall.Find(ctx, tfinstall.ExactVersion(version, installPath))
+	installDir := filepath.Join(workdir, "hcinstall")
+	err := os.MkdirAll(installDir, 0o755)
+
+	i := hcinstall.NewInstaller()
+	v := version.Must(version.NewVersion(tfVersion))
+
+	execPath, err := i.Ensure(ctx, []src.Source{
+		&fs.ExactVersion{
+			Product: product.Terraform,
+			Version: v,
+		},
+		&releases.ExactVersion{
+			Product:    product.Terraform,
+			Version:    v,
+			InstallDir: installDir,
+		},
+	})
```

Comment:
```suggestion
	})

	t.Cleanup(func() {
		if err := i.Remove(ctx); err != nil {
			t.Fatal(err)
		}
	})
```
The installer provides a method we can use to cleanup downloaded/installed files after testing.

Reply:
installDir is under the `workDir`. It's created using the `TempDir()` function which registers a cleanup function that will delete the whole `workDir`. Do you want to have another cleanup function anyway?

Reply:
You're not wrong here in that it would get cleaned anyway but the benefit of explicit cleanup function is that we can catch test failures when removal of any individual file fails and we can investigate (as that would likely indicate a bug).

Also I think that the previous `t.TempDir()` was preferable over `workdir` as calling `TempDir()` in this context would create test-specific directory which will _not_ be shared with any other test, which in turn allows us to run these tests in parallel later. Also workdir would often point to some git-versioned test data and I think it's best to avoid creating files under test in places which are not git-ignored.